settings.md
**Overview**
  - [`version`](#version)
  - [`ssid`](#ssid)
  - [`password`](#password)
  - [`channel`](#channel)
  - [`hidden`](#HIDDEN)
  - [`captivePortal`](#CAPTIVEPORTAL)
  - [`lang`](#lang)
  - [`autosave`](#autosave)
  - [`autosaveTime`](#autosave-time)
  - [`displayInterface`](#display-interface)
  - [`displayTimeout`](#DISPLAYTIMEOUT)
  - [`serialInterface`](#serial-interface)
  - [`webInterface`](#web-interface)
  - [`ledEnabled`](#ledEnabled)
  - [`maxCh`](#max-ch)
  - [`macAP`](#macap)
  - [`macSt`](#macst)
  - [`chTime`](#ch-time)
  - [`minDeauths`](#MINDEAUTHS)
  - [`attackTimeout`](#attacktimeout)
  - [`forcePackets`](#force-packets)
  - [`deauthsPerTarget`](#deauths-per-target)
  - [`deauthReason`](#deauth-reason)
  - [`beaconChannel`](#beacon-channel)
  - [`beaconInterval`](#beacon-interval)
  - [`randomTx`](#randomtx)
  - [`probesPerSSID`](#PROBESPERSSID)
  
## VERSION
`String version = VERSION;`  
Version number, i.e. `v2.0`.  
**PLEASE NOTE** that this setting can only be changed in the source code.  

## SSID
`String ssid = "mbolino";`  
SSID of access point used for the web interface (if enabled).  
The length must be between 1 and 31 characters.  

## PASSWORD
`String password = "bambolino";`  
Password of access point used for the web interface (if enabled).  
The length must be between 8 and 31 characters.  

## CHANNEL
`uint8_t channel = 1;`  
Default WiFi channel that is used when starting.  

## HIDDEN
`bool hidden = false`  
Hides the access point that is used for the web interface (if enabled).   

## CAPTIVEPORTAL
`bool captivePortal = true;`  
Enables captive portal for access point (if enabled).  

## LANG
`String lang = "en";`  
Default language for the web interface.  
Be sure the language file exists!  

## AUTOSAVE
`bool autosave = true;`  
Enables automatic saving of SSIDs, device names and settings.  

## AUTOSAVE-TIME
`uint32_t autosaveTime = 30000;`  
Time interval for automatic saving in milliseconds.  

## DISPLAY-INTERFACE
`bool displayInterface = false;`  
Enables display interface.  

## DISPLAYTIMEOUT
`uint32_t displayTimeout = 600`  
Time in seconds after which the display turns off when inactive.  
To disable the display timeout, set it to 0.  

## SERIAL-INTERFACE
`bool serialInterface = true;`  
Enables serial interface.  

## WEB-INTERFACE
`bool webInterface = false;`  
Enables web interface.  

## LEDENABLED
`bool ledEnabled = true`  
Enables the (RGB) LED feature.  

## MAX-CH
`uint8_t maxCh = 13;`  
Max channel to scan on.  
US = 11, EU = 13, Japan = 14.  
For more information click [here](https://en.wikipedia.org/wiki/List_of_WLAN_channels).  

## MACAP
`uint8_t* macAP;`  
Mac address used for the access point mode.  
Please note that the mac address will only replace the internal mac address when the accesspoint mode is enabled.  
You can set a random mac address with `set macap random`.  

## MACST
`uint8_t* macSt;`  
Mac address used for the station mode.  
Please note that the mac address will only replace the internal mac address when the station mode is enabled.  
You can set a random mac address with `set macst random`.  

## CH-TIME
`uint16_t chTime = 384;`  
Time for scanning one channel before going to the next in milliseconds (only if channel hopping is enabled).  

## MINDEAUTHS
`uint16_t minDeauths = 3`  
Minimum number of deauthentication frames when scanning to change the LED to deauth mode.  

## ATTACKTIMEOUT
`uint32_t attackTimeout = 600`  
After what amount of time (in seconds) the attack will stop automatically.  
Set it to 0 to disable it.  

## FORCE-PACKETS
`uint8_t forcePackets = 1;`  
How many attempts to send out a packet.  
Set this value higher if you want to achieve a better packet rate in a busy area.  
Be careful this setting could make the device slower or more unstable.  
**Max value is 255**.  

## DEAUTHS-PER-TARGET
`uint16_t deauthsPerTarget = 10;`  
How many deauthentication and disassociation frames are sent out for each target.  

## DEAUTH-REASON
`uint8_t deauthReason = 1;`  
The [reason code](https://www.cisco.com/assets/sol/sb/WAP371_Emulators/WAP371_Emulator_v1-0-1-5/help/Apx_ReasonCodes2.html) that is sent with the deauth frames to tell the target device why the connection will be closed.  

## BEACON-CHANNEL
`bool beaconChannel = false;`  
If enabled, will sent all beacon and probe frames on different channels when running a beacon attack.  

## BEACON-INTERVAL
`bool beaconInterval = false;`  
If set true, beacons will be sent out every second. If set to false, the interval will be 100ms. 
A longer interval means more stability and less spamming of packets, but it could take longer 
until the clients find the ssids when scanning.  

## RANDOMTX
`bool randomTX = false`  
Enables randomized transmission power for sending out beacon and probe request frames.  

## PROBESPERSSID
`uint8_t probesPerSSID = 1`  
How many probe request frames are sent for each SSID.     


esp8266_deauther/Settings.cpp
#include "Settings.h"

Settings::Settings() {
  macSt = (uint8_t*)malloc(6);
  macAP = (uint8_t*)malloc(6);
}

void Settings::load() {
  DynamicJsonBuffer jsonBuffer(4000);

  // check & read file
  String json = getJsonStr();
  checkFile(FILE_PATH, json);
  JsonObject &data = parseJSONFile(FILE_PATH, jsonBuffer);

  // VERSION
  version = data.get<String>("version");

  // AP
  if(data.containsKey(keyword(S_SSID))) setSSID(data.get<String>(keyword(S_SSID)));
  if(data.containsKey(keyword(S_PASSWORD))) setPassword(data.get<String>(keyword(S_PASSWORD)));
  if(data.containsKey(keyword(S_CHANNEL))) setChannel(data.get<uint8_t>(keyword(S_CHANNEL)));
  if(data.containsKey(keyword(S_HIDDEN))) setHidden(data.get<bool>(keyword(S_HIDDEN)));
  if(data.containsKey(keyword(S_CAPTIVEPORTAL))) setCaptivePortal(data.get<bool>(keyword(S_CAPTIVEPORTAL)));

  // GENERAL
  if(data.containsKey(keyword(S_LANG))) setLang(data.get<String>(keyword(S_LANG)));
  if(data.containsKey(keyword(S_DISPLAYINTERFACE))) setDisplayInterface(data.get<bool>(keyword(S_DISPLAYINTERFACE)));
  if(data.containsKey(keyword(S_DISPLAY_TIMEOUT))) setDisplayTimeout(data.get<uint32_t>(keyword(S_DISPLAY_TIMEOUT)));
  if(data.containsKey(keyword(S_SERIALINTERFACE))) setSerialInterface(data.get<bool>(keyword(S_SERIALINTERFACE)));
  if(data.containsKey(keyword(S_WEBINTERFACE))) setWebInterface(data.get<bool>(keyword(S_WEBINTERFACE)));
  if(data.containsKey(keyword(S_LEDENABLED))) setLedEnabled(data.get<bool>(keyword(S_LEDENABLED)));
  if(data.containsKey(keyword(S_MAXCH))) setMaxCh(data.get<uint8_t>(keyword(S_MAXCH)));
  if(data.containsKey(keyword(S_MACAP))) setMacAP(data.get<String>(keyword(S_MACAP)));
  if(data.containsKey(keyword(S_MACST))) setMacSt(data.get<String>(keyword(S_MACST)));

  // SCAN
  if(data.containsKey(keyword(S_CHTIME))) setChTime(data.get<uint16_t>(keyword(S_CHTIME)));
  if(data.containsKey(keyword(S_MIN_DEAUTHS))) setMinDeauths(data.get<uint16_t>(keyword(S_MIN_DEAUTHS)));
  
  // ATTACK
  if(data.containsKey(keyword(S_ATTACKTIMEOUT))) setAttackTimeout(data.get<uint32_t>(keyword(S_ATTACKTIMEOUT)));
  if(data.containsKey(keyword(S_FORCEPACKETS))) setForcePackets(data.get<uint8_t>(keyword(S_FORCEPACKETS)));
  if(data.containsKey(keyword(S_DEAUTHSPERTARGET))) setDeauthsPerTarget(data.get<uint16_t>(keyword(S_DEAUTHSPERTARGET)));
  if(data.containsKey(keyword(S_DEAUTHREASON))) setDeauthReason(data.get<uint8_t>(keyword(S_DEAUTHREASON)));
  if(data.containsKey(keyword(S_BEACONCHANNEL))) setBeaconChannel(data.get<bool>(keyword(S_BEACONCHANNEL)));
  if(data.containsKey(keyword(S_BEACONINTERVAL))) setBeaconInterval(data.get<bool>(keyword(S_BEACONINTERVAL)));
  if(data.containsKey(keyword(S_RANDOMTX))) setRandomTX(data.get<bool>(keyword(S_RANDOMTX)));
  if(data.containsKey(keyword(S_PROBESPERSSID))) setProbesPerSSID(data.get<uint8_t>(keyword(S_PROBESPERSSID)));
  
  if (version != VERSION){
    //reset();
    version = VERSION;
    changed = true;
  }

  prnt(S_SETTINGS_LOADED);
  prntln(FILE_PATH);

  save(true); // force saving
}

void Settings::load(String filepath){
  String tmp = FILE_PATH;
  FILE_PATH = filepath;
  load();
  FILE_PATH = tmp;
}

void Settings::reset() {
  // VERSION
  version = VERSION;
  
  // AP
  setSSID("mbolino");
  setPassword("bambolino");
  setChannel(1);
  setHidden(false);
  setCaptivePortal(true);

  // GENERAL
  setLang("en");
  setAutosave(true);
  setAutosaveTime(10000);
  setDisplayInterface(USE_DISPLAY);
  setDisplayTimeout(600);
  setSerialInterface(true);
  setWebInterface(true);
  setLedEnabled(true);
  setMaxCh(14);
  wifi_get_macaddr(0x00, macSt);
  wifi_get_macaddr(0x01, macAP);

  // SCAN
  setChTime(384);
  setMinDeauths(3);
  
  // ATTACK
  setAttackTimeout(600);
  setForcePackets(3);
  setDeauthsPerTarget(20);
  setDeauthReason(1);
  setBeaconChannel(false);
  setBeaconInterval(false);
  setRandomTX(false);
  setProbesPerSSID(1);

  prntln(S_SETTINGS_RESETED);
}

String Settings::getJsonStr() {
  DynamicJsonBuffer jsonBuffer(4000);
  JsonObject &data = jsonBuffer.createObject();

  // Version
  data.set("version", VERSION);
  
  // AP
  data.set(keyword(S_SSID), ssid);
  data.set(keyword(S_PASSWORD), password);
  data.set(keyword(S_CHANNEL), channel);
  data.set(keyword(S_HIDDEN), hidden);
  data.set(keyword(S_CAPTIVEPORTAL), captivePortal);

  // GENERAL
  data.set(keyword(S_LANG), lang);
  data.set(keyword(S_AUTOSAVE), autosave);
  data.set(keyword(S_AUTOSAVETIME), autosaveTime);
  data.set(keyword(S_DISPLAYINTERFACE), displayInterface);
  data.set(keyword(S_DISPLAY_TIMEOUT), displayTimeout);
  data.set(keyword(S_SERIALINTERFACE), serialInterface);
  data.set(keyword(S_WEBINTERFACE), webInterface);
  data.set(keyword(S_LEDENABLED), ledEnabled);
  data.set(keyword(S_MAXCH), maxCh);
  data.set(keyword(S_MACAP), macToStr(getMacAP()));
  data.set(keyword(S_MACST), macToStr(getMacSt()));

  // SCAN
  data.set(keyword(S_CHTIME), chTime);
  data.set(keyword(S_MIN_DEAUTHS), minDeauths);
  
  // ATTACK
  data.set(keyword(S_ATTACKTIMEOUT), attackTimeout);
  data.set(keyword(S_FORCEPACKETS), forcePackets);
  data.set(keyword(S_DEAUTHSPERTARGET), deauthsPerTarget);
  data.set(keyword(S_DEAUTHREASON), deauthReason);
  data.set(keyword(S_BEACONCHANNEL), beaconChannel);
  data.set(keyword(S_BEACONINTERVAL), beaconInterval);
  data.set(keyword(S_RANDOMTX), randomTX);
  data.set(keyword(S_PROBESPERSSID), probesPerSSID);
  
  String buf;
  data.printTo(buf);

  return buf;
}

void Settings::save(bool force) {
  if (force || changed) {
    String buf = getJsonStr();
    if(writeFile(FILE_PATH, buf)){
      prnt(S_SETTINGS_SAVED);
      prntln(FILE_PATH);
      changed = false;
    }else{
      prnt(F("ERROR: saving "));
      prntln(FILE_PATH);
    }
  }
}

void Settings::save(bool force, String filepath) {
  String tmp = FILE_PATH;
  FILE_PATH = filepath;
  save(force);
  FILE_PATH = tmp;
}

void Settings::print() {
  String settingsJson = getJsonStr();
  settingsJson.replace("{", "{\r\n");
  settingsJson.replace("}", "\r\n}");
  settingsJson.replace(",", "\r\n");
  prntln(S_SETTINGS_HEADER);
  prntln(settingsJson);
}

void Settings::set(const char* str, String value) {
  // booleans
  if (eqls(str,S_BEACONCHANNEL)) setBeaconChannel(eqls(value,STR_TRUE));
  else if (eqls(str,S_AUTOSAVE)) setAutosave(eqls(value,STR_TRUE));
  else if (eqls(str,S_BEACONINTERVAL)) setBeaconInterval(eqls(value,STR_TRUE));
  else if (eqls(str,S_SERIALINTERFACE)) setSerialInterface(eqls(value,STR_TRUE));
  else if (eqls(str,S_DISPLAYINTERFACE)) setDisplayInterface(eqls(value,STR_TRUE));
  else if (eqls(str,S_WEBINTERFACE)) setWebInterface(eqls(value,STR_TRUE));
  else if (eqls(str,S_RANDOMTX)) setRandomTX(eqls(value,STR_TRUE));
  else if (eqls(str,S_LEDENABLED)) setLedEnabled(eqls(value,STR_TRUE));
  else if (eqls(str,S_HIDDEN)) setHidden(eqls(value,STR_TRUE));
  else if (eqls(str,S_CAPTIVEPORTAL)) setCaptivePortal(eqls(value,STR_TRUE));

  // integer
  else if (eqls(str,S_FORCEPACKETS)) setForcePackets(value.toInt());
  else if (eqls(str,S_AUTOSAVETIME)) setAutosaveTime(value.toInt());
  else if (eqls(str,S_DEAUTHSPERTARGET)) setDeauthsPerTarget(value.toInt());
  else if (eqls(str,S_CHTIME)) setChTime(value.toInt());
  else if (eqls(str,S_MAXCH)) setMaxCh(value.toInt());
  else if (eqls(str,S_CHANNEL)) setChannel(value.toInt());
  else if (eqls(str,S_DEAUTHREASON)) setDeauthReason(value.toInt());
  else if (eqls(str,S_ATTACKTIMEOUT)) setAttackTimeout(value.toInt());
  else if (eqls(str,S_PROBESPERSSID)) setProbesPerSSID(value.toInt());
  else if (eqls(str,S_MIN_DEAUTHS)) setMinDeauths(value.toInt());
  else if (eqls(str,S_DISPLAY_TIMEOUT)) setDisplayTimeout(value.toInt());

  // strings
  else if (eqls(str,S_LANG)) setLang(value);
  else if (eqls(str,S_SSID)) setSSID(value);
  else if (eqls(str,S_PASSWORD)) setPassword(value);
  else if (eqls(str,S_MACAP)) setMacAP(value);
  else if (eqls(str,S_MACST)) setMacSt(value);
  else if (eqls(str,S_MAC) && value.equalsIgnoreCase("random")){
    setMacSt(value);
    setMacAP(value);
  }

  else if (eqls(str,S_VERSION)) prntln(S_ERROR_VERSION);

  else {
    prnt(S_ERROR_NOT_FOUND);
    prntln(str);
    return;
  }

  prnt(S_CHANGED_SETTING);
  prntln(str);
}

String Settings::get(const char* str) {
  if (eqls(str,S_SETTINGS)) print();
  // booleans
  else if (eqls(str,S_BEACONCHANNEL)) return b2s(getBeaconChannel());
  else if (eqls(str,S_AUTOSAVE)) return b2s(getAutosave());
  else if (eqls(str,S_BEACONINTERVAL)) return b2s(getBeaconInterval());
  else if (eqls(str,S_SERIALINTERFACE)) return b2s(getSerialInterface());
  else if (eqls(str,S_DISPLAYINTERFACE)) return b2s(getDisplayInterface());
  else if (eqls(str,S_WEBINTERFACE)) return b2s(getWebInterface());
  else if (eqls(str,S_RANDOMTX)) return b2s(getRandomTX());
  else if (eqls(str,S_LEDENABLED)) return b2s(getLedEnabled());
  else if (eqls(str,S_HIDDEN)) return b2s(getHidden());
  else if (eqls(str,S_CAPTIVEPORTAL)) return b2s(getCaptivePortal());

  // integer
  else if (eqls(str,S_FORCEPACKETS)) return (String)getForcePackets();
  else if (eqls(str,S_AUTOSAVETIME)) return (String)getAutosaveTime();
  else if (eqls(str,S_DEAUTHSPERTARGET)) return (String)getDeauthsPerTarget();
  else if (eqls(str,S_CHTIME)) return (String)getChTime();
  else if (eqls(str,S_ATTACKTIMEOUT)) return (String)getAttackTimeout();
  else if (eqls(str,S_MAXCH)) return (String)getMaxCh();
  else if (eqls(str,S_CHANNEL)) return (String)getChannel();
  else if (eqls(str,S_DEAUTHREASON)) return (String)getDeauthReason();
  else if (eqls(str,S_PROBESPERSSID)) return (String)getProbesPerSSID();
  else if (eqls(str,S_MIN_DEAUTHS)) return (String)getMinDeauths();
  else if (eqls(str,S_DISPLAY_TIMEOUT)) return (String)getDisplayTimeout();

  // strings
  else if (eqls(str,S_SSID)) return getSSID();
  else if (eqls(str,S_LANG)) return getLang();
  else if (eqls(str,S_PASSWORD)) return getPassword();
  else if (eqls(str,S_MACAP)) return macToStr(getMacAP());
  else if (eqls(str,S_MACST)) return macToStr(getMacSt());
  else if (eqls(str,S_MAC)) return "AP: " + macToStr(getMacAP()) + ", Station: " + macToStr(getMacSt());
  else if (eqls(str,S_VERSION)) return getVersion();

  else {
    prnt(S_ERROR_NOT_FOUND);
    prntln(str);
  }
  
  return "";
}

// ===== GETTERS ===== //
String Settings::getVersion() {
  return version;
}

uint16_t Settings::getDeauthsPerTarget() {
  return deauthsPerTarget;
}

uint8_t Settings::getDeauthReason() {
  return deauthReason;
}

bool Settings::getBeaconChannel() {
  return beaconChannel;
}

uint8_t Settings::getForcePackets() {
  return forcePackets;
}

bool Settings::getAutosave() {
  return autosave;
}

uint32_t Settings::getAutosaveTime() {
  return autosaveTime;
}

uint8_t Settings::getMaxCh() {
  return maxCh;
}

bool Settings::getBeaconInterval() {
  return beaconInterval;
}

uint8_t Settings::getChannel() {
  return channel;
}

String Settings::getSSID() {
  return ssid;
}

String Settings::getPassword() {
  return password;
}

bool Settings::getSerialInterface() {
  return serialInterface;
}

bool Settings::getDisplayInterface() {
  return displayInterface;
}

bool Settings::getWebInterface() {
  return webInterface;
}

uint16_t Settings::getChTime() {
  return chTime;
}

uint8_t* Settings::getMacSt() {
  return macSt;
}

uint8_t* Settings::getMacAP() {
  return macAP;
}

bool Settings::getRandomTX() {
  return randomTX;
}

uint32_t Settings::getAttackTimeout() {
  return attackTimeout;
}

bool Settings::getLedEnabled() {
  return ledEnabled;
}

uint8_t Settings::getProbesPerSSID() {
  return probesPerSSID;
}

bool Settings::getHidden() {
  return hidden;
}

bool Settings::getCaptivePortal() {
  return captivePortal;
}

uint16_t Settings::getMinDeauths(){
  return minDeauths;
}

uint32_t Settings::getDisplayTimeout(){
  return displayTimeout;
}

String Settings::getLang(){
  return lang;
}

// ===== SETTERS ===== //

void Settings::setDeauthsPerTarget(uint16_t deauthsPerTarget) {
  Settings::deauthsPerTarget = deauthsPerTarget;
  changed = true;
}

void Settings::setDeauthReason(uint8_t deauthReason) {
  Settings::deauthReason = deauthReason;
  changed = true;
}

void Settings::setBeaconChannel(bool beaconChannel) {
  Settings::beaconChannel = beaconChannel;
  changed = true;
}

void Settings::setForcePackets(uint8_t forcePackets) {
  if(forcePackets > 0){
    Settings::forcePackets = forcePackets;
    changed = true;
  }
}

void Settings::setAutosave(bool autosave) {
  Settings::autosave = autosave;
  changed = true;
}

void Settings::setAutosaveTime(uint32_t autosaveTime) {
  Settings::autosaveTime = autosaveTime;
  changed = true;
}

void Settings::setMaxCh(uint8_t maxCh) {
  Settings::maxCh = maxCh;
  changed = true;
}

void Settings::setBeaconInterval(bool beaconInterval) {
  Settings::beaconInterval = beaconInterval;
  changed = true;
}

void Settings::setChannel(uint8_t channel) {
  if (channel >= 1 && channel <= maxCh) {
    Settings::channel = channel;
    setWifiChannel(channel);
    changed = true;
    prnt(S_CHANNEL_CHANGE);
    prntln(channel);
  } else {
    prnt(S_CHANNEL_ERROR);
    prntln(maxCh);
  }
}

void Settings::setSSID(String ssid) {
  if (ssid.length() > 0 && ssid.length() <= 32) {
    ssid = fixUtf8(ssid);
    Settings::ssid = ssid;
    changed = true;
  } else {
    prntln(S_ERROR_SSID_LEN);
  }
}

void Settings::setPassword(String password) {
  if (password.length() >= 8 && password.length() <= 32) {
    password = fixUtf8(password);
    Settings::password = password;
    changed = true;
  } else {
    prntln(S_ERROR_PASSWORD_LEN);
  }
}

void Settings::setSerialInterface(bool serialInterface) {
  Settings::serialInterface = serialInterface;
  changed = true;
}

void Settings::setDisplayInterface(bool displayInterface) {
  Settings::displayInterface = displayInterface;
  changed = true;
}

void Settings::setWebInterface(bool webInterface) {
  Settings::webInterface = webInterface;
  changed = true;
}

void Settings::setChTime(uint16_t chTime) {
  Settings::chTime = chTime;
  changed = true;
}

void Settings::setMacSt(String macStr) {
  uint8_t mac[6];

  if(eqls(macStr, S_RANDOM))
    getRandomMac(mac);
  else
    strToMac(macStr, mac);
  
  setMacSt(mac);
}

bool Settings::setMacSt(uint8_t* macSt){
  if(macSt[0] % 2 == 0){
    memcpy(Settings::macSt, macSt, 6);
    changed = true;
    return true;
  }
  return false;
}

void Settings::setMacAP(String macStr){
  uint8_t mac[6];

  if(eqls(macStr, S_RANDOM))
    getRandomMac(mac);
  else
    strToMac(macStr, mac);
  
  setMacAP(mac);
}

bool Settings::setMacAP(uint8_t* macAP){
  if(macAP[0] % 2 == 0){
    memcpy(Settings::macAP, macAP, 6);
    changed = true;
    return true;
  }
  return false;
}

void Settings::setRandomTX(bool randomTX){
  Settings::randomTX = randomTX;
  changed = true;
}

void Settings::setAttackTimeout(uint32_t attackTimeout){
  Settings::attackTimeout = attackTimeout;
  changed = true;
}

void Settings::setLedEnabled(bool ledEnabled) {
  Settings::ledEnabled = ledEnabled;
  changed = true;
}

void Settings::setProbesPerSSID(uint8_t probesPerSSID) {
  if(probesPerSSID > 0){
    Settings::probesPerSSID = probesPerSSID;
    changed = true;
  }
}

void Settings::setHidden(bool hidden) {
  Settings::hidden = hidden;
  changed = true;
}

void Settings::setCaptivePortal(bool captivePortal){
  Settings::captivePortal = captivePortal;
  changed = true;
}

void Settings::setMinDeauths(uint16_t minDeauths){
  Settings::minDeauths = minDeauths;
  changed = true;
}

void Settings::setDisplayTimeout(uint32_t displayTimeout){
  Settings::displayTimeout = displayTimeout;
  changed = true;
}

void Settings::setLang(String lang){
  Settings::lang = lang;
  changed = true;
}


